pkg/privilege/privileges/cache.go

```diff
@@ -916,6 +918,16 @@ func (record *baseRecord) assignUserOrHost(row chunk.Row, i int, f *resolve.Resu
 
 func (p *immutable) decodeUserTableRow(row chunk.Row, fs []*resolve.ResultField) error {
 	var value UserRecord
+	defaultAuthPlugin := ""
```

Comment:
What happen if a user create some users with default auth plugin = x
and then the set the default auth variable = y?
@CbcWestwolf

Reply:
The 'plugin' field in `mysql.user` table would be filled with 'x'. So in most cases, `defaultAuthPlugin` would not attend to decode a row, unless the `plugin` field is manully cleaned by tidb user, or some bugs.